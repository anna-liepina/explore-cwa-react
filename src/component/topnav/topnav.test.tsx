import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import TopNav from './topnav';
import type { ITopNavProps } from './topnav';

const ProxyComponent: React.FC = (props: ITopNavProps) =>
    <MemoryRouter>
        <TopNav {...props} />
    </MemoryRouter>;

describe('<TopNav/>', () => {
    const props = {
    };

    const optionalProps = {
        'data-cy': 'optProps.data-cy',
        className: 'optProps.className',
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { asFragment } = render(<ProxyComponent {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it('with optional/required props', () => {
            const { asFragment } = render(<ProxyComponent {...props} {...optionalProps} />);

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
