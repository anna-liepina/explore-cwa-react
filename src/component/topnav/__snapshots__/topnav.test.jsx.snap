// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNav/> render with default/required props 1`] = `
<DocumentFragment>
  <nav
    class="topnav "
    data-cy="-topnav"
  >
    <a
      class="topnav__link"
      href="/"
    >
      <img
        alt="Logo"
        class="topnav__logo--main"
        data-cy="-topnav-logo"
        src="/assets/img/1.png"
      />
    </a>
    <img
      alt="UK postcode areas"
      class=" topnav__item"
      data-cy="topnav-postcode_tree--drawer--open"
      src="/assets/img/united-kingdom.svg"
    />
    <a
      class="topnav__link--github"
      href="//github.com/anna-liepina/explore-cwa-react"
      target="_blank"
    >
      <img
        alt="GitHub logo"
        class="topnav__logo--github"
        data-cy="-topnav-github"
        src="/assets/img/github-logo.png"
      />
    </a>
  </nav>
</DocumentFragment>
`;

exports[`<TopNav/> render with optional/required props 1`] = `
<DocumentFragment>
  <nav
    class="topnav optProps.className"
    data-cy="optProps.data-cy-topnav"
  >
    <a
      class="topnav__link"
      href="/"
    >
      <img
        alt="Logo"
        class="topnav__logo--main"
        data-cy="optProps.data-cy-topnav-logo"
        src="/assets/img/1.png"
      />
    </a>
    <img
      alt="UK postcode areas"
      class=" topnav__item"
      data-cy="topnav-postcode_tree--drawer--open"
      src="/assets/img/united-kingdom.svg"
    />
    <a
      class="topnav__link--github"
      href="//github.com/anna-liepina/explore-cwa-react"
      target="_blank"
    >
      <img
        alt="GitHub logo"
        class="topnav__logo--github"
        data-cy="optProps.data-cy-topnav-github"
        src="/assets/img/github-logo.png"
      />
    </a>
  </nav>
</DocumentFragment>
`;
